tomb-common/src/banyan_api/models/metadata.rs

```diff
@@ -84,24 +84,37 @@ impl Metadata {
                 bucket_id,
                 root_cid: root_cid.clone(),
                 metadata_cid: metadata_cid.clone(),
-                data_size,
+                expected_data_size,
                 metadata_stream,
             })
             .await?;
-        Ok((
-            Self {
-                id: response.id,
-                bucket_id,
-                root_cid,
-                metadata_cid,
-                data_size,
-                state: response.state,
-            },
-            StorageTicket {
-                host: response.storage_host,
-                authorization: response.storage_authorization,
-            },
-        ))
+        match response.storage_host {
+            None => Ok((
+                Self {
+                    id: response.id,
+                    bucket_id,
+                    root_cid,
+                    metadata_cid,
+                    data_size: 0,
+                    state: response.state,
+                },
+                None,
+            )),
+            Some(_) => Ok((
+                Self {
```

Comment:
this Self obj is the same but is defined twice; move outside the match statement for succinctness